### ISSUE TITLE
How to Collect Number of Tickets Only

### ISSUE THREAD

**Issue by abonetwork** [trepidity/autotask-samples#1]
What is the function in collecting the number of ticket?
Can you help me with this.Thanks

**Comment by trepidity** [comment]
You want to query all tickets?

Like this?

```
sb.Append ("<queryxml><entity>Ticket</entity>").Append (System.Environment.NewLine);
sb.Append ("<query><field>id<expression op=\"greaterthan\">0</expression></field></query>").Append (System.Environment.NewLine);
sb.Append ("</queryxml>").Append (System.Environment.NewLine);
```

**Comment by abonetwork** [comment]
Thanks mate.

When i used the query u give me. It only shows the dates from 2012 not the current year.
We currenlty using Autotask, Did i missed something?
Does the query have limit, it seems the length is only 500 queries?
Thank you.

**Comment by trepidity** [comment]
Yes that is true, You have to page through the results for each 500 entries.
You could also change the query to something like createDateTime greaterThan some date.

Have you seen the API Docs about creating queries?

**Comment by abonetwork** [comment]
Thank you so much

**Comment by abonetwork** [comment]
How to i make this as queryxml?
"SELECT \* from Ticket where CreatedDate >= 'Whole day ex.YYYY-MM-DD, will get all tickets within that day'  AND 'select all ticket who have not been completed' "
Thank you

**Comment by trepidity** [comment]
Yes, you can do multiple queries "conditions" together.

```
QueryXML:
<queryxml> <entity>contact</entity>
<query> <condition>
<field>firstname
<expression op=”equals”>Joe</expression>
</field> </condition>
<condition> <field>lastname
<expression op=”equals”>Smith</expression> </field>
</condition> </query>
</queryxml>
```

Just review whatever entity you want to query and then you can query on any of the fields that it supports.

**Comment by abonetwork** [comment]
Thank you very much.

**Comment by abonetwork** [comment]
Hi Mate,
How can i retrieve the ticket history of a ticket and how would i get the task of a ticket?

**Comment by trepidity** [comment]
@abon22 have you looked at the Web Services API Documentation? It basically outlines all these questions. You usually have to get the Service Call, or Resource and then track down through associations to get this information.
